### ISSUE TITLE
Dashboard sharing settings modal stays open after opened during feature tour

### ISSUE THREAD

**Issue by wpdarren** [google/site-kit-wp#6478]
## Bug Description

<details>
<summary>Original description </summary>

I have two UX/UI observations that are linked and I think could be an enhancement.  

When visiting the Dashboard for the first time, the Dashboard Sharing tooltip tour appears. 

1. On the second tip `Manage view access for other roles` there is no back link to go back to the first tip. A user could accidentally click next on the 1st tip and then not be able to go back and view it.  

2. On the second tip `Manage view access for other roles` when you click 'Got it' the tooltip disappears but the Dashboard Sharing & Permissions modal still remains.  It's possible that the majority of users would not want to set up the feature, so then have to close the modal.  

Neither of these are huge issues but thought I would create a ticket for discussion. 

![image.png](https://images.zenhubusercontent.com/5f99965247ef5a1da6985d7d/62f3c09c-3212-473f-bfbb-cba3d9a5ead1)

</details>

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When the dashboard sharing settings are opened programmatically by the feature tour, the modal should be closed again automatically when the tour is completed or dismissed

## Implementation Brief
*  Using `assets/js/feature-tours/dashboard-sharing.js`, within the `callback` function, check if the `action` is `ACTIONS.CLOSE` to close the modal. In other words, add the condition where we have the `ACTIONS.STOP` condition.

### Test Coverage
* N/A

## QA Brief

1. Set up the Site Kit without Dashboard Sharing enabled in the tester plugin
2. Once set up, go to the tester plugin and enable Dashboard sharing
3. Go to the dashboard and you’ll see the Dashboard Sharing Feature Tour
4. Go through the Dashboard Sharing Feature Tour and check that the Dashboard Sharing Settings modal closes again automatically when the tour is completed or dismissed

Note: A console error when moving from step 1 to step 2 of the tour is expected (see: https://github.com/google/site-kit-wp/issues/6478#issuecomment-1462001581), we have a new issue open to fix it: #6703.

## Changelog entry

* Fix the dashboard sharing modal not being closed after closing the feature tour issue.

**Comment by aaemnnosttv** [comment]
> When visiting the Dashboard for the first time, the Dashboard Sharing tooltip tour appears.
> 
> 1. On the second tip `Manage view access for other roles` there is no back link to go back to the first tip. A user could accidentally click next on the 1st tip and then not be able to go back and view it.

Users can go back to the first tip from the second, but only from the introductory (3-step) tour.

<img width="888" alt="image" src="https://user-images.githubusercontent.com/1621608/219493572-074b5bb6-d044-4195-af6a-8a476d12650b.png">

The dashboard sharing tour is actually two separate tours (1 for the header element, and 2 for the settings). If someone dismisses the first tour on the first step, they'll see the second tour the first time they open the sharing settings. They already know how they got there, so there's no point in showing the first step of the tour there. This was done intentionally.

See 

- https://github.com/google/site-kit-wp/issues/5382
- https://github.com/google/site-kit-wp/issues/5520

> 2\. On the second tip `Manage view access for other roles` when you click 'Got it' the tooltip disappears but the Dashboard Sharing & Permissions modal still remains.  It's possible that the majority of users would not want to set up the feature, so then have to close the modal.

This is a bug and should be fixed 👍

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@sashadoes I am testing this on develop branch (but also tested on main with same result) and as you can see from the screencast below: for the first tooltip, you click on 'Next' and then the modal opens and the 2nd tooltip appears. When I click on 'Got it' the tooltip disappears but the modal remains open.   There is a warning message saying 'Target not mounted' which appears when you click on the 'Next' link on the first tooltip and the 2nd tooltip loads. 

https://user-images.githubusercontent.com/73545194/223634182-d6ca4779-4939-44de-928d-093f32d5c581.mp4

**Comment by aaemnnosttv** [comment]
Note: While testing, I noticed another issue with the tour which I am rather certain was not a problem when it was initially done before. I'm not sure if it's fixable though from what I've seen.

The problem is that when you view the 3 step tour, go to step 2, and then back to the first, if you then try to go back to the 2nd step, the tour "crashes" (disappears) because it can't find the target for the step.

**Comment by tofumatt** [comment]
> Note: While testing, I noticed another issue with the tour which I am rather certain was not a problem when it was initially done before. I'm not sure if it's fixable though from what I've seen.
> 
> The problem is that when you view the 3 step tour, go to step 2, and then back to the first, if you then try to go back to the 2nd step, the tour "crashes" (disappears) because it can't find the target for the step.

I've noticed transitioning from the first to second step has a React Joyride no-op message as well:

<img width="1120" alt="CleanShot 2023-03-09 at 11 25 54" src="https://user-images.githubusercontent.com/90871/224009653-6caf5bb3-95df-45ea-91d6-f273aa82c22e.png">

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@tofumatt I am still not seeing the modal close when I have finished the tour as per the QAB. 

> Go through the Dashboard Sharing Feature Tour and check that the Dashboard Sharing Settings modal closes again automatically when the tour is completed or dismissed.

https://user-images.githubusercontent.com/73545194/224052298-f118c45e-41c9-4b34-a14a-3d05d09ef01b.mp4

**Comment by tofumatt** [comment]
Hi @wpdarren, thanks for that.

Unfortunately it looks like the dismissal is a lot buggier and more complicated than anticipated. @aaemnnosttv and I just spent some time debugging things and looks like we'll need to make some changes to our React Joyride setup to iron things out.

For now, nothing is super-broken when using the feature tour and we don't want to delay the release for this issue, as it will take a bit of time to fix/ship—that work will be done in #6703.

For now it is acceptable if the modal is OR is not closed when the feature tour is over. I know that isn't great but it's what we've decided is okay for this release, and until #6703 is completed. I'll adjust the ACs in #6703 to have a more consistent outcome/state.
